subsys/net/lib/lwm2m_client_utils/lwm2m/neighbour_cell_info.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <net/lwm2m.h>
+#include <net/lwm2m_client_utils.h>
+#include <lwm2m_engine.h>
+
+#include <modem/modem_info.h>
+#include <modem/lte_lc.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(lwm2m_neighbour_cell, CONFIG_LWM2M_CLIENT_UTILS_LOG_LEVEL);
+#define MAX_INSTANCE_COUNT CONFIG_LWM2M_CLIENT_UTILS_SIGNAL_MEAS_INFO_INSTANCE_COUNT
+
+#define SIGNAL_MEAS_INFO_PHYS_CELL_ID		0
+#define SIGNAL_MEAS_INFO_ECGI			1
+#define SIGNAL_MEAS_INFO_ARFCN_EUTRA		2
+#define SIGNAL_MEAS_INFO_RSRP_RESULT		3
+#define SIGNAL_MEAS_INFO_RSRQ_RESULT		4
+#define SIGNAL_MEAS_INFO_UE_RXTX_TIMEDIFF	5
+
+static void update_signal_meas_object(struct lte_lc_ncell cell, uint16_t index)
+{
+	int obj_inst_id;
+	char path[sizeof("10256/65535/0")];
+
+	obj_inst_id = lwm2m_signal_meas_info_index_to_inst_id(index);
+
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/0", obj_inst_id);
+	lwm2m_engine_set_s32(path, cell.phys_cell_id);
+	/* We don't set the resource 1 as the lte_lc_ncell struct doesn't
+	 * contain MCC and MNC for calculating ECGI
+	 */
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/2", obj_inst_id);
+	lwm2m_engine_set_s32(path, cell.earfcn);
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/3", obj_inst_id);
+	lwm2m_engine_set_s32(path, cell.rsrp - MODEM_INFO_RSRP_OFFSET_VAL);
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/4", obj_inst_id);
+	lwm2m_engine_set_s32(path, cell.rsrq);
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/5", obj_inst_id);
+	lwm2m_engine_set_s32(path, cell.time_diff);
+}
+
+static void reset_signal_meas_object(uint16_t index)
+{
+	int obj_inst_id;
+	char path[sizeof("10256/65535/0")];
+
+	obj_inst_id = lwm2m_signal_meas_info_index_to_inst_id(index);
+
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/0", obj_inst_id);
+	lwm2m_engine_set_s32(path, 0);
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/2", obj_inst_id);
+	lwm2m_engine_set_s32(path, 0);
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/3", obj_inst_id);
+	lwm2m_engine_set_s32(path, 0);
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/4", obj_inst_id);
+	lwm2m_engine_set_s32(path, 0);
+	snprintf(path, sizeof(path), "10256/%" PRIu16 "/5", obj_inst_id);
+	lwm2m_engine_set_s32(path, 0);
+}
+
+static void update_signal_meas_objects(struct lte_lc_cells_info cells)
```

Comment:
Do you really want to receive copy of the struct in the parameter, or would it be less stack use if you would just bring the pointer?